code/src/components/HabitTracker.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import styled from 'styled-components/macro';
+import '@lottiefiles/lottie-player';
+
+import { Item } from './Item';
+import { AddItem } from './AddItem';
+import { todos } from '../reducers/todos';
+
+export const HabitTracker = () => {
+	const items = useSelector((store) => store.todos.items);
+	const dispatch = useDispatch();
+
+	const completedItems = items.filter((item) => item.isCompleted);
+	const notCompletedItems = items.filter((item) => !item.isCompleted);
+
+	const handleClearNotes = () => {
+		dispatch(todos.actions.cleanItems());
+	};
+
+	return (
+		<NotesContainer>
+			<h1>Handy todo-list & habit-tracker</h1>
+
+			<Notes>
+				<AddItem />
+				{notCompletedItems.map((item) => (
+					<Item key={item.id} id={item.id} {...item} />
+				))}
+				<div
+					style={
+						completedItems.length === 0
+							? { display: 'none' }
+							: { display: 'block' }
+					}>
+					Completed posts:
+				</div>
+				{completedItems.map((item) => (
+					<Item key={item.id} id={item.id} {...item} />
+				))}
+				<Button
+					onClick={() => handleClearNotes()}
+					style={
+						completedItems.length === 0 && notCompletedItems.length === 0
+							? { display: 'none' }
+							: { display: 'block' }
+					}>
+					Clear all notes
+				</Button>
+				<lottie-player
+					autoplay
+					loop
+					mode="normal"
+					src="https://assets7.lottiefiles.com/packages/lf20_h59xofz0.json"
+					style={
+						completedItems.length < 1 && notCompletedItems.length < 1
+							? { display: 'block', width: '300px' }
+							: { display: 'none' }
+					}></lottie-player>
+				<NotesCounter
+					style={
+						items.length === 0 ? { display: 'none' } : { display: 'block' }
+					}>
+					<div>Total notes on todo-list: {items.length}</div>
+					<div>Completed notes: {completedItems.length}</div>
+					<div>Not completed notes: {notCompletedItems.length}</div>
+				</NotesCounter>
+			</Notes>
+		</NotesContainer>
+	);
+};
+
+const NotesContainer = styled.section`
```

Comment:
Well done using styled components 👏 Maybe you could add the styled components in a seperate style file to make this file a bit smaller ?